Sider of the application

leo7418 opened the pull request:
Display a left sider with three sections: "LAN", "Own Devices", and "Contacts".

leo7418 committed:
- New file Application.cpp/hpp:
    -  Display sider and content
    - On sider:
        - Display LAN, Own devices and Contacts sections
        - For each section item, display name and connected/disconnected icon
- On main:
    - Initialize an application
    - Run the application

f-michaut commented on source/Application.cpp:
```diff
@@ -0,0 +1,147 @@
+        // tgui::Panel::Ptr content = this->buildContent();
+        // content->setPosition({tgui::bindRight(sider), "0"});
+        // this->gui.add(content);
+    }
```
I feel like this should live in the constructor - or be a private method called from the constructor, doesn't really matter

f-michaut commented on source/main.cpp:
```diff
@@ -1,74 +1,17 @@
+        std::cerr << "Failed to load TGUI widgets: " << e.what() << std::endl;
     }
+    app.loop();
+    return 1;
```
Return 1 is for errors, when the app is closed normally we should return 0;

f-michaut commented on source/Application.cpp:
```diff
@@ -0,0 +1,147 @@
+                std::cout << "HIDE" << std::endl;
+                sider->moveWithAnimation({ "-width", "0" }, sf::milliseconds(ANIMATION_DURATION));
+                button->resizeWithAnimation({ "64", button->getSize().y }, sf::milliseconds(ANIMATION_DURATION));
+
+                /*
+                unsigned int signalId = sider->onAnimationFinish([=]() {
+                    sider->setVisible(false);
+                    sider->onAnimationFinish.disconnect(signalId);
+                    std::cout << "Animation finished with id " << signalId << std::endl;
+                });
+                /*/
```
Let's clean the code `std::cout` and commented-out code if they serve no purpose

leo7418 committed:
- On Application:
    - Move init function inside constructor
    - Clean code
- On main:
    - Return 1 only for errors

f-michaut commented on include/Application.hpp:
```diff
@@ -0,0 +1,36 @@
+#include <TGUI/Backend/SFML-Graphics.hpp>
+
+namespace FileShare::GUI {
+    #define ANIMATION_DURATION 250
```
let's use constants (const or constexpr - static variables) instead of macros. Macros leak outisde of namespaces.

f-michaut reviewed (approved):
All good, minor comment
